micromanager_gui/_core_widgets/_exposure_widget.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+from qtpy import QtWidgets as QtW
+from qtpy.QtCore import Qt
+from superqt.utils import signals_blocked
+
+if TYPE_CHECKING:
+    from pymmcore_plus import CMMCorePlus
+
+from pymmcore import g_Keyword_CoreCamera, g_Keyword_CoreDevice
+
+from micromanager_gui import _core
+
+
+class ExposureWidget(QtW.QWidget):
+    def __init__(
+        self,
+        camera: str = None,
+        *,
+        parent: Optional[Qt.QWidget] = None,
+        core: Optional[CMMCorePlus] = None,
+    ):
+        super().__init__()
+        self._mmc = core or _core.get_core_singleton()
+        self._camera = camera or self._mmc.getCameraDevice()
+
+        self.label = QtW.QLabel()
+        self.label.setText(" ms")
+        self.label.setMaximumWidth(30)
+        self.spinBox = QtW.QDoubleSpinBox()
+        self.spinBox.setAlignment(Qt.AlignCenter)
+        self.spinBox.setMinimum(1.0)
+        self.spinBox.setMaximum(100000.0)
+        self.spinBox.setKeyboardTracking(False)
+        layout = QtW.QHBoxLayout()
+        layout.addWidget(self.spinBox)
+        layout.addWidget(self.label)
+        self.setLayout(layout)
+
+        self._on_load()
+        self._mmc.events.exposureChanged.connect(self._on_exp_changed)
+        self._mmc.events.systemConfigurationLoaded.connect(self._on_load)
+
+        self.spinBox.valueChanged.connect(self._mmc.setExposure)
+
+    def setCamera(self, camera: str = None):
+        """
+        Set which camera this widget tracks
+
+        Parameters
+        ----------
+        camera : str
+            The camera device label. If None then use the current Camera device.
+        """
+        orig_cam = self._camera
+        self._camera = camera or self._mmc.getCameraDevice()
+        if orig_cam != self._camera:
+            self._on_load()
+
+    def _on_load(self):
+        with signals_blocked(self.spinBox):
+            if self._camera and self._camera in self._mmc.getLoadedDevices():
+                self.setEnabled(True)
+                self.spinBox.setValue(self._mmc.getExposure(self._camera))
+            else:
+                self.setEnabled(False)
+
+    def _on_exp_changed(self, camera: str, exposure: float):
+        if camera == self._camera:
+            with signals_blocked(self.spinBox):
+                self.spinBox.setValue(exposure)
+
+    def _on_exp_set(self, exposure: float):
+        self._mmc.setExposure(self._camera, exposure)
+
+
+class DefaultCameraExposureWidget(ExposureWidget):
+    def __init__(
+        self, *, parent: Optional[Qt.QWidget] = None, core: Optional[CMMCorePlus] = None
+    ):
+        super().__init__(core=core)
+        self._mmc.events.devicePropertyChanged(
+            g_Keyword_CoreDevice, g_Keyword_CoreCamera
+        ).connect(self._camera_updated)
+
+    def setCamera(self, camera: str = None, force: bool = False):
```

Comment:
are you thinking of a use case for this?  or just maybe nice to have?  maybe "yagni"?

Reply:
Do you mean the method on `ExposureWidget` or the version on the DefaultCamera version?

Reply:
The one with the big warning on default cam... ah, but maybe I see now, it's already there in the superclass and you're just restricting it on the subclass?

Reply:
yeah exactly.